### DISCUSSION TITLE
What's the best way to use Type ORM in Nuxt 3?

### DISCUSSION THREAD

**Discussion by renanleandrof** [nuxt/nuxt#16836] [Questions]
Hi, new to Nuxt and haven't fully undertood how to structure my code.

Where should i put my Type ORM files?
Should i use Plugins or Modules or something else?

To use Type ORM i need to export a Datasource and i'm not sure where to put it.
Thanks!

**Comment by StevenTaing** [comment]
I also used to try it. I put TypeORM in plugin and when I start server, the auto migration is working, but it's not working in server api. I saw the tutorial of configure Prisma in Nuxt 3, yet still wonder how to configure TypeORM in Nuxt 3. If anyone found the way please share. thanks...

**Comment by alee** [comment]
We also tried adding it as a server side plugin (e.g., `plugins/typeorm.server.ts`) that [provides](https://v3.nuxtjs.org/guide/directory-structure/plugins/#automatically-providing-helpers) access to the typeorm `DataSource` or `Manager` but couldn't get it to infer Column types, with the nitro dev server raising `ColumnTypeUndefinedError` for our entities on startup.

I think it has something to do with esbuild not supporting typescript decorators and `emitDecoratorMetadata` but didn't have the time / resources to dig further into that rabbit hole and decided to wait until someone else figures it out :sweat_smile: 

https://github.com/virtualcommons/port-of-mars/issues/809
